crates/krun/src/bin/krun.rs

```diff
@@ -206,10 +207,25 @@ fn main() -> Result<()> {
         "PATH",                        // needed by `krun-guest` program
         "RUST_LOG",
     ];
+
+    // https://github.com/AsahiLinux/docs/wiki/Devices
+    const ASAHI_DEVS: [&str; 12] = [
+        "t8103", "t6000", "t6001", "t6002", "t8112", "t6020", "t6021", "t6022", "t8121", "t6030",
+        "t6031", "t6034",
+    ];
     for key in WELL_KNOWN_ENV_VARS {
         let value = match env::var(key) {
             Ok(value) => value,
             Err(VarError::NotPresent) => {
+                if key == "MESA_LOADER_DRIVER_OVERRIDE" {
+                    for compatible in
+                        read_to_string("/proc/device-tree/compatible")?.split(&[',', '\0'][..])
+                    {
+                        if ASAHI_DEVS.iter().any(|&s| s == compatible) {
```

Comment:
I don't think this would match? For example, it's `t8112apple` on my machine.

```
teohhanhui@han-macbook-air:~$ cat /proc/device-tree/compatible
apple,j413apple,t8112apple,arm-platform
```

Reply:
Oh...

```
teohhanhui@han-macbook-air:~/Projects/FEX-Emu/RootFS$ od -c /proc/device-tree/compatible
0000000   a   p   p   l   e   ,   j   4   1   3  \0   a   p   p   l   e
0000020   ,   t   8   1   1   2  \0   a   p   p   l   e   ,   a   r   m
0000040   -   p   l   a   t   f   o   r   m  \0
0000052
```

But I think that means we should match on `t8112\0apple` instead?

Reply:
Having the `'\0'` in the middle of the string looks kind strange.

Reply:
I found it confusing at first, but it's a genius thing for them to do... The users who simply `cat /proc/device-tree/compatible` would see it as `t8112apple`. But there's a hidden NUL byte for programs to parse, in case they want to separate the different components.

Reply:
It took me way too long to find out that the NUL was in the middle and not at the end.

Reply:
You know what... I should have read the docs. I was completely mistaken. :laughing:

https://docs.kernel.org/devicetree/usage-model.html

So on my machine it's actually:

"apple,j413", "apple,t8112", "apple,arm-platform"

The NUL byte (`\0`) is the separator.

Reply:
Ok, that makes more sense.

Reply:
Fixed it.